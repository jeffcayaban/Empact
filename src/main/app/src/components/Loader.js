import React from 'react'
import BarLoader from "react-spinners/BarLoader";

/**
 * Is a loader that will be displayed whenever data is being fetched.
 */

class Loader extends React.PureComponent {
    render() {
        return (
            <div className={'loaderContainerStyle'}>
                <div className={'centerStyle'}>
                    <BarLoader color={'#7d7d7d'} loading={this.props.isLoading} />
                </div>
            </div>
        );
    }
}

export default Loader;
