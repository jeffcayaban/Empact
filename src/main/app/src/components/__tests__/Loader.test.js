import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Loader from "../Loader";

describe('Loader', () => {
    it('should render correctly', () => {
        const component = shallow(<Loader />);
        expect(toJson(component)).toMatchSnapshot();
    });
});