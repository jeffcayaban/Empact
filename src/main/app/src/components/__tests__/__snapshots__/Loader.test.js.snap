// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
<div
  className="loaderContainerStyle"
>
  <div
    className="centerStyle"
  >
    <onlyUpdateForKeys(Loader)
      className=""
      color="#7d7d7d"
      height={4}
      heightUnit="px"
      loading={true}
      width={100}
      widthUnit="px"
    />
  </div>
</div>
`;
